nixos/modules/services/backup/tarsnap.nix

```diff
@@ -242,9 +254,11 @@ in
 
     systemd.services."tarsnap@" = {
       description = "Tarsnap archive '%i'";
-      requires    = [ "network.target" ];
+      requires    = [ "network-online.target" ];
+      after       = [ "network-online.target" ];
 
-      path = [ pkgs.tarsnap pkgs.coreutils ];
+      path = [ pkgs.tarsnap pkgs.coreutils pkgs.iputils ];
+      preStart = "/run/current-system/sw/bin/sh -c 'while ! ping -q -c 1 betatest-server.tarsnap.com &> /dev/null; do true; done'";
```

Comment:
Without this, running the service when offline or when the tarsnap server is otherwise unavailable might count as having performed a backup (just guessing here).

I dislike this. I prefer using after/requires network-online.target and leave it at that.